### ISSUE TITLE
Add Young tableau support to graph plugin

### ISSUE THREAD

**Issue by hammerfunctor** [XmacsLabs/mogan#1239]
See the patch
```
diff --color -ruN /home/huzf/test/mogan/plugins/tmpy/graph/ytableau.py tmpy/graph/ytableau.py
--- /home/huzf/test/mogan/plugins/tmpy/graph/ytableau.py	1969-12-31 19:00:00.000000000 -0500
+++ tmpy/graph/ytableau.py	2023-10-10 06:19:28.080040934 -0400
@@ -0,0 +1,57 @@
+#!/usr/bin/env python
+###############################################################################
+##
+## MODULE      : ytableau.py
+## DESCRIPTION : Young Tableau support
+## COPYRIGHT   : (C) 2019  Darcy Shen, Massimiliano Gubinelli
+##                   2023  Hammer
+##
+## This software falls under the GNU general public license version 3 or later.
+## It comes WITHOUT ANY WARRANTY WHATSOEVER. For details, see the file LICENSE
+## in the root directory or <http://www.gnu.org/licenses/gpl-3.0.html>.
+
+import os
+from subprocess import Popen, PIPE, STDOUT
+from .latex import LaTeX
+from ..protocol import *
+
+
+class YTableau(LaTeX):
+    def __init__(self, name = "yt"):
+        super(YTableau, self).__init__()
+        self.name = name
+
+        self.pre_code = """
+\\documentclass{standalone}
+\\usepackage{ytableau}
+\\usepackage{amsmath}
+\\usepackage{amssymb}
+\\begin{document}
+"""
+        self.post_code = """
+\\end{document}
+"""
+        self.message = "TeXmacs interface to ytableau"
+        self.default_width = "0.05par"
+
+    def available(self):
+        if not super(YTableau, self).available():
+            return False
+        for sty in ("standalone", "tikz"):
+            if len (super(YTableau, self).kpsewhich(sty + ".sty")) <= 0:
+                flush_err ("Failed to find " + sty +".sty,"
+                           " please install the missing LaTeX packages\n")
+                return False
+        return True
+        
+    def evaluate(self, code):
+        if not (code.lstrip().startswith("\\documentclass")):
+            if code.lstrip().startswith("\\ytableausetup"):
+                pass
+            elif not (code.lstrip().startswith("\\begin{ytableau}")) and \
+                 not (code.lstrip().startswith("\\ydiagram")) and \
+                 not (code.lstrip().startswith("\\begin{multline}")):
+                code = "\\begin{ytableau}\n" + code + "\n\\end{ytablaeu}"
+            code = self.pre_code + "\n" + code + "\n" + self.post_code
+
+        super(YTableau, self).evaluate(code)
diff --color -ruN /home/huzf/test/mogan/plugins/tmpy/session/tm_graph.py tmpy/session/tm_graph.py
--- /home/huzf/test/mogan/plugins/tmpy/session/tm_graph.py	2023-10-10 06:15:41.556702723 -0400
+++ tmpy/session/tm_graph.py	2023-10-09 17:32:12.783388639 -0400
@@ -43,6 +43,7 @@
 from tmpy.graph.pdflatex  import PDFLaTeX
 from tmpy.graph.gnuplot   import Gnuplot
 from tmpy.graph.diagrams  import Diagrams
+from tmpy.graph.ytableau  import YTableau
 from tmpy.compat          import *
 
 #import logging as log
@@ -98,7 +99,7 @@
 grapvizs = list(map(lambda x: Graphviz(x), ["dot", "neato", "twopi", "circo", "fdp", "sfdp", "patchwork", "osage"]))
 diagrams = list(map(lambda x: Diagrams(x), ["flowchart", "sequence", "railroad"]))
 others = [Asymptote(), PlantUML(), Mermaid(), XYpic(), TikZ(),
-          FeynMF(), LaTeX(), PDFLaTeX(), Gnuplot(), DraTeX()]
+          FeynMF(), LaTeX(), PDFLaTeX(), Gnuplot(), DraTeX(), YTableau()]
 graphs = list(filter(lambda x: x.available(), grapvizs + diagrams + others))
 graph_names = list(map(lambda x: x.name, graphs))
 ```

**Comment by darcy-shen** [comment]
Please create pull requests to https://github.com/texmacs/plugins-in-python

**Comment by darcy-shen** [comment]
Python-implemented plugins will be removed in Mogan v1.2.0, because we are planning to put plugins in plugin center.
